components/brave_wallet_ui/ledger/ledger.ts

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import TransportWebHID from '@ledgerhq/hw-transport-webhid'
+import Sol from '@ledgerhq/hw-app-solana'
+
+import {
+  LedgerCommand,
+  UnlockCommand,
+  UnlockResponse,
+  UnlockResponsePayload,
+  GetAccountCommand,
+  GetAccountResponse,
+  GetAccountResponsePayload,
+  SignTransactionCommand,
+  SignTransactionResponsePayload,
+  SignTransactionResponse,
+} from '../common/hardware/ledgerjs/ledger-messages'
+
+import { addLedgerCommandHandler } from '../common/hardware/ledgerjs/ledger-command-handler'
+
+const createUnlockResponse = (command: UnlockCommand, result: boolean, error?: any): UnlockResponsePayload => {
```

Comment:
avoid `any`

Reply:
I think we could probably use https://github.com/LedgerHQ/ledger-live/tree/develop/libs/ledgerjs/packages/errors

My impression is that hardware device error handling probably needs another pass back to front.  Some outstanding issues I see -
* A lot of errors are not actually being bubbled up to the UI (e.g. all ledger hardware device errors when adding accounts)
* Error handling is not consistent between chrome://wallet and chrome://wallet-panel
* Conflation of various categories of error

Let me know what you think, but I think I'd rather iterate on those changes in subsequent pull requests